### ISSUE TITLE
sharing with circle requires https

### ISSUE THREAD

**Issue by schiessle** [nextcloud/circles#242]
I just tried circles 0.15.1 with Nextcloud 14.0.1, created a personal circle and then tried to share a folder with the circle. Then I get a 403 with this error message: "cURL error 7: Failed to connect to localhost port 443: Connection refused (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)"

But if I reload the browser the share is there. So it seems like we create the share correctly but at some point in the process we try to make a https request with curl to the local server which is setup without ssl on my development system.

**Comment by daita** [comment]
yes, this is the broadcast of the shares, this should be improved, but you can:

`./occ config:app:set circles --value 1 allow_non_ssl_links` 
`./occ config:app:set circles --value 1 local_is_non_ssl`

**Comment by fgsl** [comment]
@schiessle and @daita, if I understand, the scenario is development, because if it is production, it is recommended that server has a SSL certificate. Then, I think that if is debug mode is on, settings allow_non_ssl_links and local_is_non_ssl could be assumed as on too.

Related methods of OCA\Circles\Service\ConfigService could be changed so:

```
	public function getLocalAddress() {
		return (($this->isLocalNonSSL() || \OC::$server->getConfig()->getSystemValue('debug',false) ? 'http://' : '')
			   . $this->request->getServerHost();
	}
```
```
	public function isNonSSLLinksAllowed() {
		if ($this->allowedNonSSLLinks === -1) {
			$this->allowedNonSSLLinks = 
				(int)$this->getAppValue(self::CIRCLES_ALLOW_NON_SSL_LINKS);
		}
                $this->allowedNonSSLLinks = (int)($this->allowedNonSSLLinks || \OC::$server->getConfig()->getSystemValue('debug',false));
		return ($this->allowedNonSSLLinks === 1);
	}
```
What do you think about it?

**Comment by wiswedel** [comment]
resolved by #322
